### ISSUE TITLE
YouTube authentication doesn't let user choose a specific channel in Google Account

### ISSUE THREAD

**Issue by WangleLine** [Tyrrrz/YoutubeDownloader#375]
### Version

1.10.3

### Platform

Windows 11

### Steps to reproduce

- Log into google account with multiple channels
- Pass url of a private video from one of the non-default channels into youtube downloader
- receive error message

### Details

I'm trying to download private videos from my own channel. When I log into my google account via youtube downloader, it automatically switches to a different youtube channel from that google account and I have no further control over this.

The private download fails both with the regular link, as well as when shared via the "share" button beneath the video

![ImageGlass_yPAjRJf8LH](https://github.com/Tyrrrz/YoutubeDownloader/assets/34343336/2fe0394e-4d67-4639-a163-9b83c309c823)

![ImageGlass_Thm6Ulrld4](https://github.com/Tyrrrz/YoutubeDownloader/assets/34343336/faad1fdb-1bf1-4ef2-a046-831277895681)

### Checklist

- [X] I have looked through existing issues to make sure that this bug has not been reported before
- [X] I have provided a descriptive title for this issue
- [X] I have made sure that that this bug is reproducible on the latest version of the application
- [X] I have provided all the information needed to reproduce this bug as efficiently as possible
- [ ] I have sponsored this project

**Comment by Tyrrrz** [comment]
Can you try this build and see if it works? https://github.com/Tyrrrz/YoutubeDownloader/actions/runs/5494961866

**Comment by WangleLine** [comment]
oh god I'm really sorry, I completely missed your response until now - Could I get a new download link for the build?

**Comment by Tyrrrz** [comment]
It's okay. I believe that implementation still chose the account for you, it just always chose the first one.

We need a way to let the user make the choice during the authentication sequence. I'm not sure how yet though.

**Comment by WangleLine** [comment]
Ok! Thank you. Please let me know if I can help somehow

**Comment by SnorreSelmer** [comment]
I just authenticated yesterday, and I could choose between my three "identities" on my YouTube account just fine.

**Comment by WangleLine** [comment]
@SnorreSelmer What version were you on?

**Comment by SnorreSelmer** [comment]
1.10.7

**Comment by Tyrrrz** [comment]
> I just authenticated yesterday, and I could choose between my three "identities" on my YouTube account just fine.

Were you provided with an option to select the right identity as soon as you logged in?

**Comment by SnorreSelmer** [comment]
Yes. It asked which account to use, I chose my only account, and then it asked which identity to use (I have three).

**Comment by Tyrrrz** [comment]
Interesting. @WangleLine do you get the same?

**Comment by WangleLine** [comment]
I just reinstalled YoutubeDownloader (1.10.8), logged into youtube and wasn't given the option to choose between channels. I have no idea which channel it picked to be honest
